src/OpenTelemetry.Extensions.Hosting/OpenTelemetryBuilder.cs

```diff
@@ -68,6 +68,29 @@ internal OpenTelemetryBuilder(IServiceCollection services)
         return this;
     }
 
+    /// <summary>
+    /// Registers an action to configure the <see cref="ResourceBuilder"/>s used
+    /// by tracing and metrics.
+    /// </summary>
+    /// <remarks><inheritdoc cref="ConfigureResource(System.Action{OpenTelemetry.Resources.ResourceBuilder})" path="/remarks"/></remarks>
+    /// <param name="configure"><see cref="ResourceBuilder"/> configuration
+    /// action.</param>
+    /// <returns>The supplied <see cref="OpenTelemetryBuilder"/> for chaining
+    /// calls.</returns>
+    public OpenTelemetryBuilder ConfigureResource(
```

Comment:
This API I think might work. I want to take it for a spin and try a couple things but I don't see any obvious reasons why it would be trouble to have this.

nit: One thing though, would you reorder the params so it is `IServiceProvider, ResourceBuilder` I think in all our "factory" overloads `IServiceProvider` comes first.

Reply:
> nit: One thing though, would you reorder the params so it is IServiceProvider, ResourceBuilder I think in all our "factory" overloads IServiceProvider comes first.

Well, the idea is that it make a lot more sense that the optional parameter comes second so you can just add it, but I mean the ordering doesn't make a huge difference me...